examples/chat-with-mdns/main.go

```diff
@@ -116,6 +116,9 @@ func main() {
 	peerChan := initMDNS(host, cfg.RendezvousString)
 
 	peer := <-peerChan // will block untill we discover a peer
+	for peer.ID == host.ID() {
```

Comment:
Why? That would mean that we only discover ourselves, or am I missing something here?

Reply:
Hello, thank you for your reply.

My understanding is that we first take a connection from this peerchan. When the ID of this connection is the same as that of our current host, we will continue to get it from peerchan. Until we found another peer with a different ID. We jump out of the for loop.

In fact, similar codes have appeared in https://github.com/libp2p/go-libp2p/blob/master/examples/chat-with-rendezvous/chat.go （line161）you have written. See the first two lines of code in the for loop.

(ps: maybe you should run my modified code and unmodified code)

looking forward to your reply.
 
```go
go func() {
		for peer := range peerChan {
			if peer.ID == host.ID() {
				continue
			}
			fmt.Println("Found peer:", peer)

			fmt.Println("Connecting to:", peer)
			stream, err := host.NewStream(ctx, peer.ID, protocol.ID(config.ProtocolID))

			if err != nil {
				fmt.Println("Connection failed:", err)
				continue
			} else {
				rw := bufio.NewReadWriter(bufio.NewReader(stream), bufio.NewWriter(stream))

				go writeData(rw)
				go readData(rw)
			}

			fmt.Println("Connected to:", peer)
		}
	}()
 ```

Reply:
![image](https://user-images.githubusercontent.com/49364055/180672353-d813b312-ce82-436c-8065-2d164f7a8040.png)
This is a screenshot. First, I compiled the code, and then tried to run the code in two terminals. I found that both of them found themselves and tried to establish a connection with themselves.

But I don't think it should happen that I try to connect with myself for the first time. So when we find a peer with the same ID as ourselves, we should not try to connect, but continue to wait.

![image](https://user-images.githubusercontent.com/49364055/180672795-dbb7e834-6088-4c42-8408-a50066547eae.png)
After compiling my modified code, this program can successfully create a connection with another peer every time. Then the two peers can communicate.

looking forward to your reply.

Reply:
I got the logic the wrong way round, sorry for that!

I'm wondering if this fix should live inside our mDNS code though. We _shouldn't_ discover our own node.

Reply:
```go
for _, info := range infos {
	if info.ID == s.host.ID() {
		continue
	}
	go s.notifee.HandlePeerFound(info)
}
```
Hello, thank you for your reply.

Do you mean here? Add such code in line 179 of [mdns.go](https://github.com/libp2p/go-libp2p/blob/master/p2p/discovery/mdns/mdns.go) file.

Or should we not find ourselves in the discovery stage?

looking forward to your reply.

Reply:
Or do you just want me to modify mdns.go under the chat-with-mDNS file directory? 
looking forward to your reply.

Reply:
Can you please send a separate PR for your changes to mdns.go? It's an actual bug fix in libp2p!

Reply:
Thank you for your reply. I have resubmitted another PR.